### DISCUSSION TITLE
[API] I'm getting a 404 error with the Start Push Publishing API.

### DISCUSSION THREAD

**Discussion by Cheondongmin** [OvenMediaLabs/OvenMediaEngine#1535] [Help] [answered]
Bug Description
Encountered a 404 error while using the Start Push API.

Steps to Reproduce
Entered application and vhost names in Server.xml.
Attempted to use the startPush API to broadcast an RTMP video stream.
Made a POST request to http://my_ip:8081/v1/vhosts/test_host/apps/test_app:startPush

{
"id": "test_2024_02_33_24_45",
"stream": {
"name": "test_stream",
"variantNames": ["h264_fhd", "aac"]
},
"protocol": "rtmp",
"url":"rtmp://list_ip/live_app",
"streamKey":"1080"
}

Received a 404 error, indicating the operation was unsuccessful.

{
"message": "[HTTP] Could not find application: [test_host/test_app] (404)",
"statusCode": 404
}

Expected Behavior
Expected the RTMP video to be pushed and broadcast under the specified stream name.

Logs
Logs indicate a 404 error due to the application [test_host/test_app] not being found.

Server Information
OS: CentOS 7
OvenMediaEngine Version: v0.16.4
Additional Context
Other APIs like "ScheduledChannel API" and metadata transmission APIs like "Send Event" work correctly, but the Start Push API does not.

Additional Question
Is it possible to use this API to alternately broadcast live and VOD content on the same stream? The plan is to fetch live RTMP video from another source, stop the push, then create and push to a stream playing VOD content via OME API. The question is whether the transition between videos can be seamless and if such a method is feasible.

I hope you have a great day, and thank you for your assistance.

**Comment by getroot** [comment]
Since this is not an issue, I moved it to discussion. And could you please rewrite it in English? Thank you for respecting our community rules.

**Comment by Cheondongmin** [comment]
Bug Description
Encountered a 404 error while using the Start Push API.

Steps to Reproduce
Entered application and vhost names in Server.xml.
Attempted to use the startPush API to broadcast an RTMP video stream.
Made a POST request to http://my_ip:8081/v1/vhosts/test_host/apps/test_app:startPush 

{
  "id": "test_2024_02_33_24_45",
  "stream": {
    "name": "test_stream",
    "variantNames": ["h264_fhd", "aac"]
  },
  "protocol": "rtmp",
  "url":"rtmp://list_ip/live_app",
  "streamKey":"1080"
}

Received a 404 error, indicating the operation was unsuccessful.

{
  "message": "[HTTP] Could not find application: [test_host/test_app] (404)",
  "statusCode": 404
}

Expected Behavior
Expected the RTMP video to be pushed and broadcast under the specified stream name.

Logs
Logs indicate a 404 error due to the application [test_host/test_app] not being found.

Server Information
OS: CentOS 7
OvenMediaEngine Version: v0.16.4
Additional Context
Other APIs like "ScheduledChannel API" and metadata transmission APIs like "Send Event" work correctly, but the Start Push API does not.

Additional Question
Is it possible to use this API to alternately broadcast live and VOD content on the same stream? The plan is to fetch live RTMP video from another source, stop the push, then create and push to a stream playing VOD content via OME API. The question is whether the transition between videos can be seamless and if such a method is feasible.

I hope you have a great day, and thank you for your assistance.

**Comment by getroot** [comment, marked as answer]
You have not enabled RTMPPush in your application.
https://airensoft.gitbook.io/ovenmediaengine/push-publishing#configuration

**Reply by Cheondongmin** [reply]
I made a mistake by not thoroughly reviewing the document before proceeding with the work.

I am ashamed of myself!

Thank you so much for taking the time to help me.

Have a great day!
